src/Core/SharedKernels/Enumerator/Enumerator/ViewModels/InterviewDetails/Questions/CategoricalComboboxAutocompleteViewModel.cs

```diff
@@ -58,10 +61,43 @@ public void InitFilter(string initialFilter = null)
         private List<OptionWithSearchTerm> autoCompleteSuggestions = new List<OptionWithSearchTerm>();
         public List<OptionWithSearchTerm> AutoCompleteSuggestions
         {
-            get => this.autoCompleteSuggestions;
+            get => this.autoCompleteSuggestions ??= GetLazySuggestions(FilterText);
             set => this.SetProperty(ref this.autoCompleteSuggestions, value);
         }
-        
+
+        private bool loading;
+        public bool Loading
+        {
+            get => loading;
+            set => SetProperty(ref loading, value);
+        }
+        private List<OptionWithSearchTerm> GetLazySuggestions(string filterText)
+        {
+            Task.Factory.StartNew(async () =>
+            {
+                try
+                {
+                    Loading = true;
+                    var suggestions = GetSuggestions(filterText);
+                
+                    await mvxMainThreadDispatcher.ExecuteOnMainThreadAsync(() =>
+                    {
+                        AutoCompleteSuggestions = suggestions;

```

Comment:
Could you please either introduce IsDisposed property, set it in dispose and check before updating?
After returning to dashboard no need to update.
Or could you have CancellationTokenSource and call Cancel of the task if there is no need to wait any more.

Reply:
Done